packages/xo-server/src/xapi-stats.js

```diff
@@ -347,103 +308,91 @@ export default class XapiStats {
       )
     }
 
+    const currentTimeStamp = await getServerTimestamp(xapi, host.$ref)
+
     // Limit the number of http requests
-    const hostUuid = host.uuid
-
-    if (
-      !(
-        vmUuid !== undefined &&
-        get(this._statsByObject, [vmUuid, step]) === undefined
-      ) &&
-      get(this._statsByObject, [hostUuid, step, 'localTimestamp']) + step >
-        getCurrentTimestamp()
-    ) {
-      return this._getStats(hostUuid, step, vmUuid)
+    const stats = this._getCachedStats(uuid, step, currentTimeStamp)
+    if (stats !== undefined) {
+      return stats
     }
 
-    const timestamp = await this._getNextTimestamp(xapi, host, step)
-    const json = await this._getJson(xapi, host, timestamp, step)
+    const maxDuration = step * RRD_POINTS_PER_STEP[step]
+
+    // To avoid exceed the maxDuration, we add a step
+    const optimumTimestamp = currentTimeStamp - maxDuration + step
+    const json = await this._getJson(xapi, host, optimumTimestamp, step)
     if (json.meta.step !== step) {
       throw new FaultyGranularity(
         `Unable to get the true granularity: ${json.meta.step}`
       )
     }
 
-    // It exists data
-    if (json.data.length !== 0) {
-      // Warning: Sometimes, the json.xport.meta.start value does not match with the
-      // timestamp of the oldest data value
-      // So, we use the timestamp of the oldest data value !
-      const startTimestamp = json.data[json.meta.rows - 1].t
-      const endTimestamp = get(this._statsByObject, [
-        hostUuid,
-        step,
-        'endTimestamp',
-      ])
-
-      const statsOffset = endTimestamp - startTimestamp + step
-      if (endTimestamp !== undefined && statsOffset > 0) {
-        const parseOffset = statsOffset / step
-        // Remove useless data
-        // Note: Older values are at end of json.data.row
-        json.data.splice(json.data.length - parseOffset)
-      }
-
-      // It exists useful data
-      if (json.data.length > 0) {
-        // reorder data
-        json.data.reverse()
-        forEach(json.meta.legend, (legend, index) => {
-          const [, type, uuid, metricType] = /^AVERAGE:([^:]+):(.+):(.+)$/.exec(
-            legend
-          )
-
-          const metrics = STATS[type]
-          if (metrics === undefined) {
-            return
+    if (json.data.length > 0) {
+      // reorder data
```

Comment:
What does that mean?
Which order?
Why?

Reply:
The recent values are in the top of the array. Should i not trust the RRD sorting and sort them using `Array.sort()`?

Reply:
It's not what I'm saying, but the comment does not provide any valuable info, it should explain why you are doing this.